types/office-js/index.d.ts

```diff
@@ -1021,7 +1021,100 @@ declare namespace Office {
+        /**
+         * Calls the Azure Active Directory V 2.0 endpoint to get an access token to your add-in's web application. Enables  add-ins to identify users.
+         * Server side code can use this token to access Microsoft Graph for the add-in's web application by using the
+         * {@link https://docs.microsoft.com/azure/active-directory/develop/active-directory-v2-protocols-oauth-on-behalf-of | "on behalf of" OAuth flow}.
+         *
+         * Important: In Outlook, this API is not supported if the add-in is loaded in an Outlook.com or Gmail mailbox.
+         *
+         * @remarks
+         *
+         * **Hosts**: Excel, OneNote, Outlook, PowerPoint, Word
+         *
+         * **Requirement set**: {@link https://docs.microsoft.com/office/dev/add-ins/reference/requirement-sets/identity-api-requirement-sets | IdentityAPI}
+         *
+         * This API requires a single sign-on configuration that bridges the add-in to an Azure application. Office users sign-in with Organizational
+         * Accounts and Microsoft Accounts. Microsoft Azure returns tokens intended for both user account types to access resources in the Microsoft Graph.
+         *
+         * @param callback - Optional. Accepts a callback method that can use parse the token for the user's ID or use the token in the "on behalf of" flow to get access to Microsoft Graph.
+         *                   If AsyncResult.status is "succeeded", then AsyncResult.value is the raw AAD v. 2.0-formatted access token.
+         */
+        getAccessTokenAsync(callback?: (result: AsyncResult<string>) => void): void;
+    }
+    /**
+     * Provides options for the user experience when Office obtains an access token to the add-in from AAD v. 2.0 with the getAccessTokenAsync method.
+     */
+    interface AuthOptions {
+        /**
+         * Allows Office to get an access token silently or through interactive consent, if one is required.
+         * If set to false, Office will silently try to get an access token. If it fails to do so, Office will return a descriptive error.
+         * If set to true, Office will show an interactive consent UI after it fails to silently get an access token.
+         * The prompt will only allow consent to the AAD profile scope, not to any Microsoft Graph scopes.
+         */
+        allowConsentPrompt?: boolean;
+        /**
+         * Allows Office to get an access token silently provided consent is present or show interactive UI to sign in the user.
+         * If set to false, office will silently try to get an access token. If it fails to do so, Office will return a descriptive error.
+         * If set to true, Office will show an interactive sign-in UI after it fails to silently get an access token.
+         */
+        allowSignInPrompt?: boolean;
+        /**
+         * Deprecated:
+         * Prompts the user to add their Office account (or to switch to it, if it is already added).
```

Comment:
"@deprecated" tag should note what has replaced this property. See [deprecated documentation](https://api-extractor.com/pages/tsdoc/tag_deprecated/).
```suggestion
         * Prompts the user to add their Office account (or to switch to it, if it is already added). Default value is `false`.
         *
         * @deprecated
```